tests/test_androidmodule_ctypes_finder.py

```diff
@@ -0,0 +1,127 @@
+
+from mock import MagicMock
+import os
+import shutil
+import sys
+import tempfile
+
+sys.modules["jnius"] = MagicMock()  # fake jnius used later
+
+
+# Import the tested android._ctypes_library_finder module,
+# making sure android._android won't crash us!
+# (since android._android is android-only / not compilable on desktop)
+android_module_folder = os.path.abspath(os.path.join(
+    os.path.dirname(__file__),
+    "..", "pythonforandroid", "recipes", "android", "src"
+))
+sys.path.insert(0, android_module_folder)
+sys.modules['android._android'] = MagicMock()
+import android._ctypes_library_finder
+sys.path.remove(android_module_folder)
+
+
+def _get_autoclass_with_fake_activity(returning_test_path="/testpath",
+                                      return_as_service=False):
+    """
+    Helper functions to mock autoclass() from jnius.
+
+    This will return a mocked activity "org.kivy.android.PythonActivity"
+    which supports .getApplicationContext().getPackagemanager(),
+    with that returned object supporting .getApplicationInfo()
+    for PackageManager's GET_SHARED_LIBRARY_FILES info.
+
+    (That is the functionality the android._ctypes_library_finder module
+     uses to scan the application-specific library directories)
+    """
+
+    # This is a closure to bind the value of "returning_test_path":
+    def autoclass_returning_graphical_activity(name):
+        activity = MagicMock()
+        activity.getApplicationContext().getPackageName = (
+             lambda: "com.example.testpackage"
+        )
```

Comment:
it's fine to use lambda, but mock also has a built'in feature to do what you're doing here:
```python
activity.getApplicationContext().getPackageName.return_value = "com.example.testpackage"
```

Reply:
oh neat! don't think it'd be actually shorter here though, so should I change it? not sure if lambda is considered ugly here :grimacing: I really haven't seen much mock'ing code

Reply:
Lambda is completely fine :+1:

Reply:
Anyway with the magic of `MagicMock` you don't have to define this. So you could skip these lines

Reply:
it's skipped now in the revamped test except in the one where I actually check for it